AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,72 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+
+  // Mark the node for layout in the next layout pass
+  [self invalidateCalculatedLayout];
+  
+  // Hook for subclasses to get size invalidation changes
+  [self didInvalidateSize];
+  
+  if (_supernode) {
+    ASDisplayNode *supernode = _supernode;
+    __instanceLock__.unlock();
+    // Cause supernode's layout to be invalidated
+    // We need to release the lock to prevent a deadlock
+    [supernode invalidateSize];
+    return;
+  }
+  
+  // We are now at the root node trigger a layout pass
+  [self setNeedsLayout];
+  
+  __instanceLock__.unlock();
+}
+
+- (void)didInvalidateSize
+{
+    
+}
+
+- (void)sizeToFit
```

Comment:
This seems like a good method to experiment with - we should probably have it in the API eventually.

What do you think about pulling this aside into a separate PR?  I think we'd want to explore what the right constrainedSize behavior is.

It looks like the logic will work and so we could add this quite quickly if it proves useful, but might be worth reducing complexity into two stages?